### PR TITLE
Add hooks to the hiddenRows and hiddenColumns plugins.

### PR DESCRIPTION
**Pull request by jansiegel**
Add the following hooks:
- beforeHideRows
- afterHideRows
- beforeUnhideRows
- afterUnhideRows
- beforeHideColumns
- afterHideColumns
- beforeUnhideColumns
- afterUnhideColumns
 #163

### CODE REVIEW

**Comment by AMBudnik** [comment]
I'm so happy to see this PR @jansiegel !

**Comment by wszymanski** [comment]
Please add extra test related to not calling `after...` hook when `before...` hook return `false`.

File: src/plugins/hiddenColumns/hiddenColumns.js [hunk 1]
```diff
@@ -220,6 +255,17 @@ class HiddenColumns extends BasePlugin {
     return this.hiddenColumns.indexOf(columnIndex) > -1;
   }
 
+  /**
+   * Check whether all of the provided column indexes are within the bounds of the table.
+   *
+   * @param {Array} columns Array of column indexes.
+   */
+  isColumnDataValid(columns) {
+    const outOfBounds = columns.find(column => (column < 0 || column > this.hot.countCols() - 1));
```
Comment (wszymanski) [inline src/plugins/hiddenColumns/hiddenColumns.js]: You may use also [every](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/every), but decission about using it is up to you.

**Review by wszymanski — DISMISSED** [review]

**Review by wszymanski — APPROVED** [review]
<3

**Comment by jansiegel** [comment]
Moved to https://github.com/handsontable/handsontable/pull/5651

**Pull request closed without merging**